flixel/addons/tile/FlxTilemapExt.hx

```diff
@@ -311,103 +334,116 @@ class FlxTilemapExt extends FlxTilemap
+	 * @return  Whether there were overlaps, or if a callback was specified, whatever the return value of the callback was.
 	 */
-	override public function overlapsWithCallback(Object:FlxObject, ?Callback:FlxObject->FlxObject->Bool, FlipCallbackParams:Bool = false,
-			?Position:FlxPoint):Bool
+	override public function overlapsWithCallback(object:FlxObject, ?callback:FlxObject->FlxObject->Bool, flipCallbackParams:Bool = false,
+			?position:FlxPoint):Bool
 	{
 		var results:Bool = false;
 
-		var X:Float = x;
-		var Y:Float = y;
+		var xPos:Float = x;
+		var yPos:Float = y;
 
-		if (Position != null)
+		if (position != null)
 		{
-			X = Position.x;
-			Y = Position.y;
+			xPos = position.x;
+			yPos = position.y;
+			position.putWeak();
 		}
 
 		// Figure out what tiles we need to check against
-		var selectionX:Int = Math.floor((Object.x - X) / tileWidth);
-		var selectionY:Int = Math.floor((Object.y - Y) / tileHeight);
-		var selectionWidth:Int = selectionX + (Math.ceil(Object.width / tileWidth)) + 1;
-		var selectionHeight:Int = selectionY + Math.ceil(Object.height / tileHeight) + 1;
+		var selectionX:Int = Math.floor((object.x - xPos) / scaledTileWidth);
+		var selectionY:Int = Math.floor((object.y - yPos) / scaledTileHeight);
+		var selectionWidth:Int = selectionX + Math.ceil(object.width / scaledTileWidth) + 1;
+		var selectionHeight:Int = selectionY + Math.ceil(object.height / scaledTileHeight) + 1;
 
 		// Then bound these coordinates by the map edges
-		selectionX = FlxMath.maxInt(selectionX, 0);
-		selectionY = FlxMath.maxInt(selectionY, 0);
-		selectionWidth = FlxMath.minInt(selectionWidth, widthInTiles);
-		selectionHeight = FlxMath.minInt(selectionHeight, heightInTiles);
+		selectionX = Std.int(FlxMath.bound(selectionX, 0, widthInTiles));
+		selectionY = Std.int(FlxMath.bound(selectionY, 0, heightInTiles));
+		selectionWidth = Std.int(FlxMath.bound(selectionWidth, 0, widthInTiles));
+		selectionHeight = Std.int(FlxMath.bound(selectionHeight, 0, heightInTiles));
 
-		// Then loop through this selection of tiles and call FlxObject.separate() accordingly
+		// Then loop through this selection of tiles
 		var rowStart:Int = selectionY * widthInTiles;
-		var row:Int = selectionY;
 		var column:Int;
 		var tile:FlxTile;
 		var overlapFound:Bool;
-		var deltaX:Float = X - last.x;
-		var deltaY:Float = Y - last.y;
+		var deltaX:Float = xPos - last.x;
+		var deltaY:Float = yPos - last.y;
 
-		while (row < selectionHeight)
+		for (row in selectionY...selectionHeight)
 		{
 			column = selectionX;
 
 			while (column < selectionWidth)
 			{
 				overlapFound = false;
-				tile = _tileObjects[_data[rowStart + column]];
 
-				if (tile.allowCollisions != 0)
+				var index:Int = rowStart + column;
+				if (index < 0 || index > _data.length - 1)
+				{
+					column++;
+					continue;
+				}
+
+				var dataIndex:Int = _data[index];
+				if (dataIndex < 0)
+				{
+					column++;
+					continue;
+				}
```

Comment:
why check this? how can someone cause this issue?

Reply:
rather than checking here, we should log an error or warning when invalid data is put in a tilemap